misc/history/src/utils.ts

```diff
@@ -0,0 +1,68 @@
+import { RegisterPayload, GetMessagesPayload, GetMessagesResponse } from "./types";
+import fetch from "isomorphic-unfetch";
+import type { ICore } from "@walletconnect/types";
+
+export const DEFAULT_HISTORY_URL = "https://history.walletconnect.com";
+
+export const DAY_IN_MS = 86400 * 1000;
+
+export class HistoricalMessages {
+  public constructor(private core: ICore, public messageResponse: GetMessagesResponse) {}
+
+  public injectIntoRelayer() {
+    for (const message in this.messageResponse.messages) {
+      this.core.relayer.provider.events.emit("payload", message);
+    }
+  }
+}
+
+export class HistoryClient {
+  private jwt = "";
+
+  public constructor(private core: ICore, private readonly historyUrl = DEFAULT_HISTORY_URL) {}
+
+  public async registerTags(payload: RegisterPayload, historyUrl = DEFAULT_HISTORY_URL) {
+    try {
+      await fetch(`${historyUrl}/register`, {
+        method: "POST",
+        body: JSON.stringify(payload),
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${await this.getJwt()}`
+        },
+      });
+
+    } catch (e) {
+      throw new Error("Failed to register tags");
+    }
+  }
+
+  public async getMessages(payload: GetMessagesPayload, historyUrl = DEFAULT_HISTORY_URL) {
+    const entries = Object.entries(payload).map(([key, value]) => [key, value.toString()]);
+    const params = new URLSearchParams(entries);
+
+    try {
+      const url = `${historyUrl}/messages?${params.toString()}`
+      const rs: GetMessagesResponse = await (
+        await fetch(url, {
+          method: "GET",
+          headers: {
+            "Content-Type": "application/json",
+          },
+        })
+      ).json();
+
+      return new HistoricalMessages(this.core, rs);
+    } catch (e) {
+      throw new Error(`Failed to fetch messages` );
```

Comment:
This message feels too generic to be traceable later on and isn't logging the concrete error atm, e.g.:

```suggestion
      throw new Error(`[HistoryClient] Failed to fetch messages: ${JSON.stringify(e)}`);
```